### DISCUSSION TITLE
debug poetry in vscode

### DISCUSSION THREAD

**Discussion by noamagiv** [brndnmtthws/thetagang#434] [General]
Hello,
I'm having problems debugging the code in vs-code, I followed several guides on how to run poetry in debug from vscode and none worked for me. does anyone work with vscode ? if yes, can you please share the lunch.json? thanks!

**Comment by brndnmtthws** [comment]
Can you elaborate a little on what you're trying to do? Poetry is there mostly for installing dependencies and managing the virtual environment (virtualenv).

VS Code should work with virtualenv, but you need to make sure poetry makes the venv accessible to VS Code with the following:

```shell
poetry config virtualenvs.in-project true
```

^^^ Running that will put the venv within the project dir, which VS Code will pick up. See https://python-poetry.org/docs/configuration/#virtualenvsin-project for details.

**Comment by noamagiv** [comment]
Hi, Thanks for your reply.
I'm running it from cli with the following args:
poetry run thetagang -c /home/noam/thetagang_conf/thetagang.toml --without-ibc
For learning purposes, I want to run it from vscode in debug mode, should I add main function to main.py / thetagang.py and trigger it from there? or should I create lunch.json for args? 
I enable poetry venv and select it as my interpreter. 

Thanks Again.

**Reply by brndnmtthws** [reply]
If you have the virtualenv loaded, you should be able to just run it like any normal python script. VS Code will also let you simply add `poetry run thetagang -c /home/noam/thetagang_conf/thetagang.toml --without-ibc` as a launch command if you prefer. There are some suggestions here as well: https://github.com/python-poetry/poetry/issues/5354

**Reply by noamagiv** [reply]
Thanks. 
Ill run it without poetry, I didn't find any valid way to call poetry with my argument in launch.json.
